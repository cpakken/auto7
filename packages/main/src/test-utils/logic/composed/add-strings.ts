import { LogicComposedSnapshotIn } from "@main/controllers/logic"

export const ADDSTRING: LogicComposedSnapshotIn = {
  _id: "ADDSTRING",
  composition: {
    blocks: [{ _id: "A", logicKey: "_std_ADDSTRING" }],
    paths: [
      { _id: "pA", from: { node: "in1" }, to: { block: "A", node: "in1" } },
      { _id: "pB", from: { node: "in2" }, to: { block: "A", node: "in2" } },
      { _id: "pC", from: { block: "A", node: "out" }, to: { node: "out1" } },
    ],
  },
  info: {
    label: "ADDSTRING COMPOSED",
    inputs: [
      { _id: "in1", label: "in1", typeKey: "_std_string" },
      { _id: "in2", label: "in2", typeKey: "_std_number" },
    ],
    outputs: [{ _id: "out1", label: "out1", typeKey: "_std_string" }],
  },
}

export const ADDSTRING_NESTED: LogicComposedSnapshotIn = {
  _id: "ADDSTRING_NESTED",
  composition: {
    blocks: [{ _id: "A", logicKey: "ADDSTRING" }],
    paths: [
      { _id: "pA", from: { node: "in1" }, to: { block: "A", node: "in1" } },
      { _id: "pB", from: { node: "in2" }, to: { block: "A", node: "in2" } },
      { _id: "pC", from: { block: "A", node: "out1" }, to: { node: "out1" } },
    ],
  },
  info: {
    label: "ADDSTRING NESTED",
    inputs: [
      { _id: "in1", label: "in1", typeKey: "_std_string" },
      { _id: "in2", label: "in2", typeKey: "_std_number" },
    ],
    outputs: [{ _id: "out1", label: "out1", typeKey: "_std_string" }],
  },
}

export const ADDSTRING_COMPOUND: LogicComposedSnapshotIn = {
  _id: "ADDSTRING_COMPOUND",
  composition: {
    blocks: [
      { _id: "A", logicKey: "ADDSTRING", xy: [0, 0] },
      { _id: "B", logicKey: "ADDSTRING_NESTED", xy: [0, 0] },
      { _id: "C", logicKey: "_std_ADDSTRING", xy: [0, 0] },
    ],
    paths: [
      { _id: "p0", from: { node: "in1" }, to: { block: "A", node: "in1" } },
      { _id: "p1", from: { node: "in2" }, to: { block: "A", node: "in2" } },
      { _id: "p2", from: { node: "in2" }, to: { block: "B", node: "in1" } },
      { _id: "p3", from: { node: "in3" }, to: { block: "B", node: "in2" } },
      { _id: "p4", from: { block: "A", node: "out1" }, to: { block: "C", node: "in1" } },
      { _id: "p5", from: { block: "B", node: "out1" }, to: { block: "C", node: "in2" } },
      { _id: "p6", from: { block: "C", node: "out" }, to: { node: "out1" } },
    ],
  },
  info: {
    label: "ADDSTRING COMPOUND",
    inputs: [
      { _id: "in1", label: "in1", typeKey: "_std_string" },
      { _id: "in2", label: "in2", typeKey: "_std_string" },
      { _id: "in3", label: "in3", typeKey: "_std_string" },
    ],
    outputs: [{ _id: "out1", label: "out1", typeKey: "_std_string" }],
  },
}
