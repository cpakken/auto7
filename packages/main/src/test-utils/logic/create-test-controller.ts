import { objectToArray } from "@utils/collection-fns"
import { LogicController, ILogicController, LogicNativeSnapshotIn, LogicComposedSnapshotIn } from "@main/controllers/logic"
import { createLogicNativeStoreSnapshot, createLogicComposedStoreSnapshot } from "./snapshot-factories"

import * as nativeSnapshots from "./native/snapshots"
import * as composedSnapshots from "./composed"

export function createTestLogicController(env?): ILogicController {
  const nativeLoaders: LogicNativeSnapshotIn[] = objectToArray(nativeSnapshots)
  const native = createLogicNativeStoreSnapshot(nativeLoaders)

  const composedLoaders: LogicComposedSnapshotIn[] = objectToArray(composedSnapshots)
  const composed = createLogicComposedStoreSnapshot(composedLoaders)

  const snapshot = { composed, native }

  //TODO process cachedKeys
  return LogicController.create(snapshot, env)
}
