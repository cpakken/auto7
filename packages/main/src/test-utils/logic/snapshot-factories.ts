import { LogicNativeSnapshotIn, LogicComposedSnapshotIn, ILogicComposedStoreSnapshotIn } from "@main/controllers/logic"

export function createLogicComposedStoreSnapshot(
  snapshots: LogicComposedSnapshotIn | LogicComposedSnapshotIn[]
): ILogicComposedStoreSnapshotIn {
  return (Array.isArray(snapshots) ? snapshots : [snapshots]).reduce(
    (a, snapshot) => {
      const { _id, info, composition } = snapshot
      a.info.push({ _id, ...info })
      a.composition.push({ _id, ...composition })
      return a
    },
    { info: [] as any[], composition: [] as any[] }
  )
}

export function createLogicNativeStoreSnapshot(
  snapshots: LogicNativeSnapshotIn | LogicNativeSnapshotIn[]
): ILogicComposedStoreSnapshotIn {
  return (Array.isArray(snapshots) ? snapshots : [snapshots]).reduce(
    (a, snapshot) => {
      const { _id, info } = snapshot
      a.info.push({ _id, ...info })
      return a
    },
    { info: [] as any[] }
  )
}

export function addIdsToSnapshot(snapshot: LogicComposedSnapshotIn) {
  const { _id, info, composition } = snapshot
  return {
    _id,
    info: { _id, ...info },
    composition: { _id, ...composition },
  }
}
