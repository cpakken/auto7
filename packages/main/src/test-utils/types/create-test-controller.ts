import { objectToArray } from "@utils/collection-fns"
import { ITypeController, TypeController, TypeNativeSnapshotIn } from "@main/controllers/types"
import { createTypeNativeStoreSnapshot } from "./snapshot-factories"

import * as nativeSnapshots from "./native/std.config"

export function createTestTypeController(env?): ITypeController {
  const nativeLoaders: TypeNativeSnapshotIn[] = objectToArray(nativeSnapshots)
  const native = createTypeNativeStoreSnapshot(nativeLoaders)

  // const composedLoaders: LogicComposedSnapshotIn[] = objectToArray(composedSnapshots)
  // const composed = createLogicComposedStoreSnapshot(composedLoaders)

  const snapshot = {
    native,
    // composed,
  }

  //TODO process cachedKeys
  return TypeController.create(snapshot, env)
}
