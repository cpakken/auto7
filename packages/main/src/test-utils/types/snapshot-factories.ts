import { ITypeComposedStoreSnapshotIn, TypeNativeSnapshotIn } from "@main/controllers/types"

// export function createLogicComposedStoreSnapshot(
//   snapshots: LogicComposedSnapshotIn | LogicComposedSnapshotIn[]
// ): ILogicComposedStoreSnapshotIn {
//   return (Array.isArray(snapshots) ? snapshots : [snapshots]).reduce(
//     (a, snapshot) => {
//       const { _id, info, composition } = snapshot
//       a.info.push({ _id, ...info })
//       a.composition.push({ _id, ...composition })
//       return a
//     },
//     { info: [] as any[], composition: [] as any[] }
//   )
// }

export function createTypeNativeStoreSnapshot(
  snapshots: TypeNativeSnapshotIn | TypeNativeSnapshotIn[]
): ITypeComposedStoreSnapshotIn {
  return (Array.isArray(snapshots) ? snapshots : [snapshots]).reduce(
    (a, snapshot) => {
      const { _id, info } = snapshot
      a.info.push({ _id, ...info })
      return a
    },
    { info: [] as any[] }
  )
}

// export function addIdsToSnapshot(snapshot: LogicComposedSnapshotIn) {
//   const { _id, info, composition } = snapshot
//   return {
//     _id,
//     info: { _id, ...info },
//     composition: { _id, ...composition },
//   }
// }
