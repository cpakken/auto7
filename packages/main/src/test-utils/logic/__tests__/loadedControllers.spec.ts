import { getSnapshot } from "mobx-state-tree"
import { ADDSTRING_COMPOUND } from "../composed"
import { createLogicComposedStoreSnapshot } from "../snapshot-factories"
import { createTestLogicController } from "../create-test-controller"

describe("loadedControllers", () => {
  test("convertLogicSnapshot", () => {
    const snapshot = createLogicComposedStoreSnapshot(ADDSTRING_COMPOUND)
    expect(snapshot).toMatchSnapshot()
  })
  test("createLoadedController", () => {
    const logicController = createTestLogicController()
    expect(getSnapshot(logicController)).toMatchSnapshot()
  })
})
