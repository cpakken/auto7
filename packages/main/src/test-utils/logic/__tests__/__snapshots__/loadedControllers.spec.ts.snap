// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadedControllers convertLogicSnapshot 1`] = `
Object {
  "composition": Array [
    Object {
      "_id": "ADDSTRING_COMPOUND",
      "blocks": Array [
        Object {
          "_id": "A",
          "logicKey": "ADDSTRING",
          "xy": Array [
            0,
            0,
          ],
        },
        Object {
          "_id": "B",
          "logicKey": "ADDSTRING_NESTED",
          "xy": Array [
            0,
            0,
          ],
        },
        Object {
          "_id": "C",
          "logicKey": "_std_ADDSTRING",
          "xy": Array [
            0,
            0,
          ],
        },
      ],
      "paths": Array [
        Object {
          "_id": "p0",
          "from": Object {
            "node": "in1",
          },
          "to": Object {
            "block": "A",
            "node": "in1",
          },
        },
        Object {
          "_id": "p1",
          "from": Object {
            "node": "in2",
          },
          "to": Object {
            "block": "A",
            "node": "in2",
          },
        },
        Object {
          "_id": "p2",
          "from": Object {
            "node": "in2",
          },
          "to": Object {
            "block": "B",
            "node": "in1",
          },
        },
        Object {
          "_id": "p3",
          "from": Object {
            "node": "in3",
          },
          "to": Object {
            "block": "B",
            "node": "in2",
          },
        },
        Object {
          "_id": "p4",
          "from": Object {
            "block": "A",
            "node": "out1",
          },
          "to": Object {
            "block": "C",
            "node": "in1",
          },
        },
        Object {
          "_id": "p5",
          "from": Object {
            "block": "B",
            "node": "out1",
          },
          "to": Object {
            "block": "C",
            "node": "in2",
          },
        },
        Object {
          "_id": "p6",
          "from": Object {
            "block": "C",
            "node": "out",
          },
          "to": Object {
            "node": "out1",
          },
        },
      ],
    },
  ],
  "info": Array [
    Object {
      "_id": "ADDSTRING_COMPOUND",
      "inputs": Array [
        Object {
          "_id": "in1",
          "label": "in1",
          "typeKey": "_string",
        },
        Object {
          "_id": "in2",
          "label": "in2",
          "typeKey": "_string",
        },
        Object {
          "_id": "in3",
          "label": "in3",
          "typeKey": "_string",
        },
      ],
      "label": "ADDSTRING COMPOUND",
      "outputs": Array [
        Object {
          "_id": "out1",
          "label": "out1",
          "typeKey": "_string",
        },
      ],
    },
  ],
}
`;

exports[`loadedControllers createLoadedController 1`] = `
Object {
  "composed": Object {
    "composition": Array [
      Object {
        "_id": "ADDSTRING",
        "blocks": Array [
          Object {
            "_id": "A",
            "logicKey": "_std_ADDSTRING",
            "xy": Array [
              0,
              0,
            ],
          },
        ],
        "paths": Array [
          Object {
            "_id": "pA",
            "from": Object {
              "node": "in1",
            },
            "to": Object {
              "block": "A",
              "node": "in1",
            },
          },
          Object {
            "_id": "pB",
            "from": Object {
              "node": "in2",
            },
            "to": Object {
              "block": "A",
              "node": "in2",
            },
          },
          Object {
            "_id": "pC",
            "from": Object {
              "block": "A",
              "node": "out",
            },
            "to": Object {
              "node": "out1",
            },
          },
        ],
      },
      Object {
        "_id": "ADDSTRING_NESTED",
        "blocks": Array [
          Object {
            "_id": "A",
            "logicKey": "ADDSTRING",
            "xy": Array [
              0,
              0,
            ],
          },
        ],
        "paths": Array [
          Object {
            "_id": "pA",
            "from": Object {
              "node": "in1",
            },
            "to": Object {
              "block": "A",
              "node": "in1",
            },
          },
          Object {
            "_id": "pB",
            "from": Object {
              "node": "in2",
            },
            "to": Object {
              "block": "A",
              "node": "in2",
            },
          },
          Object {
            "_id": "pC",
            "from": Object {
              "block": "A",
              "node": "out1",
            },
            "to": Object {
              "node": "out1",
            },
          },
        ],
      },
      Object {
        "_id": "ADDSTRING_COMPOUND",
        "blocks": Array [
          Object {
            "_id": "A",
            "logicKey": "ADDSTRING",
            "xy": Array [
              0,
              0,
            ],
          },
          Object {
            "_id": "B",
            "logicKey": "ADDSTRING_NESTED",
            "xy": Array [
              0,
              0,
            ],
          },
          Object {
            "_id": "C",
            "logicKey": "_std_ADDSTRING",
            "xy": Array [
              0,
              0,
            ],
          },
        ],
        "paths": Array [
          Object {
            "_id": "p0",
            "from": Object {
              "node": "in1",
            },
            "to": Object {
              "block": "A",
              "node": "in1",
            },
          },
          Object {
            "_id": "p1",
            "from": Object {
              "node": "in2",
            },
            "to": Object {
              "block": "A",
              "node": "in2",
            },
          },
          Object {
            "_id": "p2",
            "from": Object {
              "node": "in2",
            },
            "to": Object {
              "block": "B",
              "node": "in1",
            },
          },
          Object {
            "_id": "p3",
            "from": Object {
              "node": "in3",
            },
            "to": Object {
              "block": "B",
              "node": "in2",
            },
          },
          Object {
            "_id": "p4",
            "from": Object {
              "block": "A",
              "node": "out1",
            },
            "to": Object {
              "block": "C",
              "node": "in1",
            },
          },
          Object {
            "_id": "p5",
            "from": Object {
              "block": "B",
              "node": "out1",
            },
            "to": Object {
              "block": "C",
              "node": "in2",
            },
          },
          Object {
            "_id": "p6",
            "from": Object {
              "block": "C",
              "node": "out",
            },
            "to": Object {
              "node": "out1",
            },
          },
        ],
      },
    ],
    "info": Array [
      Object {
        "_id": "ADDSTRING",
        "inputs": Array [
          Object {
            "_id": "in1",
            "label": "in1",
            "typeKey": "_string",
          },
          Object {
            "_id": "in2",
            "label": "in2",
            "typeKey": "_number",
          },
        ],
        "keysCache": undefined,
        "label": "ADDSTRING COMPOSED",
        "outputs": Array [
          Object {
            "_id": "out1",
            "label": "out1",
            "typeKey": "_string",
          },
        ],
      },
      Object {
        "_id": "ADDSTRING_NESTED",
        "inputs": Array [
          Object {
            "_id": "in1",
            "label": "in1",
            "typeKey": "_string",
          },
          Object {
            "_id": "in2",
            "label": "in2",
            "typeKey": "_number",
          },
        ],
        "keysCache": undefined,
        "label": "ADDSTRING NESTED",
        "outputs": Array [
          Object {
            "_id": "out1",
            "label": "out1",
            "typeKey": "_string",
          },
        ],
      },
      Object {
        "_id": "ADDSTRING_COMPOUND",
        "inputs": Array [
          Object {
            "_id": "in1",
            "label": "in1",
            "typeKey": "_string",
          },
          Object {
            "_id": "in2",
            "label": "in2",
            "typeKey": "_string",
          },
          Object {
            "_id": "in3",
            "label": "in3",
            "typeKey": "_string",
          },
        ],
        "keysCache": undefined,
        "label": "ADDSTRING COMPOUND",
        "outputs": Array [
          Object {
            "_id": "out1",
            "label": "out1",
            "typeKey": "_string",
          },
        ],
      },
    ],
  },
  "native": Object {
    "info": Array [
      Object {
        "_id": "_std_ADDSTRING",
        "inputs": Array [
          Object {
            "_id": "in1",
            "label": "in1",
            "typeKey": "_string",
          },
          Object {
            "_id": "in2",
            "label": "in2",
            "typeKey": "_string",
          },
        ],
        "label": "ADDSTRING NATIVE",
        "outputs": Array [
          Object {
            "_id": "out",
            "label": "out",
            "typeKey": "_string",
          },
        ],
      },
    ],
  },
}
`;
